### DISCUSSION TITLE
Integration of AirSim Plugin into Carla simulator

### DISCUSSION THREAD

**Discussion by Tomakko** [microsoft/AirSim#4301] [Support Q&A]
Hi,

i was wondering if it is possible to integrate the AirSim into carla, i.e. make the AirSim UE plugin coexist with the carla UE plugin.

I made a first attempt do do so. However, i need to start CarlaGameMode instead of AirSimGameMode and so the drone does now show up.

I am new to Unreal. Are the GameModes just entry points to the plugins' code (then integration would be feasible, i guess) or does a lot of stuff depend on those (which would likely mean that the approach is doomed).

Thanks!

**Comment by rajat2004** [comment]
This sounds interesting, I haven't tried it myself but can try to help out.
You can see the [AirSimGameMode.cpp](https://github.com/microsoft/AirSim/blob/master/Unreal/Plugins/AirSim/Source/AirSimGameMode.cpp) file, doesn't do much apart from setting up static classes like `ASimHUD` which will actually start up the loading of settings, etc. Maybe you can try putting this piece of code somewhere appropriate in Carls? I guess you don't want the popup to select the car or multirotor, so having a settings.json should avoid that popup. I don't have much idea about what goes into the blueprint side of things though

**Comment by fabrizioschiano** [comment]
@Tomakko did you manage to do any advancement?
